### ISSUE TITLE
Investigate CSV "required" stanza to install dependent operators

### ISSUE THREAD

**Issue by kaczyns** [kabanero-io/kabanero-operator#40]
The CSV allows you to list "required" CRDs.  The documentation is not clear about how these should be processed.  Our initial test seems to show that OLM will ignore any CSV that required CRDs, even if they are installed.  This should be investigated further.  Is it that:

1) The Operator will not be read from the catalog until all its required CRDs are defined?
2) The Operator will not be read from the catalog unless its required CRSs were installed by OLM?
3) The Operator will not be read from the catalog unless the Operator which owns the required CRDs is available to be installed in/by some other available catalog (such as `operator-marketplace`)?

This may involve testing on Open Shift 4 to realize the expected behavior, and then coming back to OKD 3.11 to see if it can be duplicated there.

The ideal behavior would be for the user to click on the Kabanero operator in the catalog, and have the dependent operators (Tekton Pipeline, Knative Eventing and Knative Serving) be installed automatically if they are not already installed.  This may not be possible, but, we'd like to understand what the actual behavior is.

**Comment by tspewak** [comment]
In OKD 311 I managed to get the dependent operators (Tekton Pipeline, Knative Eventing and Knative Serving) to be installed automatically by packaging the kabanaro, Tekton Pipeline, Knative Eventing and Knative Serving together so I can have a CatalogSource yaml point to the docker image.
In ODK clicking on the Kabanero operator in the catalog does not work. I get a blank screen.
I had to create and apply OperatorGroup, CatalogSource and Subscription yamls.

I added the following to the kabanero csv under customresourcedefinitions: following owned:
    required:
    - name: knativeservings.serving.knative.dev
      version: v1alpha1
      kind: KnativeServing
      displayName: Knative Serving
      description: Knative Serving
    - name: knativeeventings.eventing.knative.dev
      version: v1alpha1
      kind: KnativeEventing
      displayName: Knative Eventing
      description: Knative Eventing
    - name: installs.tekton.dev
      version: v1alpha1
      kind: Install
      displayName: OpenShift Pipelines Install
      description: OpenShift Pipelines using Tekton

I had to update the Tekton Pipeline csv to get it to work.
In the installModes: section I had to change AllNamespaces to supported: true example
  - supported: true
    type: AllNamespaces

Here are the yaml files I created:
apiVersion: operators.coreos.com/v1alpha2
kind: OperatorGroup
metadata:
  name: kabanero-operator-group
  namespace: kabanero

apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: kabanero-operator-catalog-source
  namespace: openshift-operator-lifecycle-manager
spec:
  sourceType: grpc
  image: spewak/kabanero:latest
  displayName: "Kabanero Operator"
  publisher: IBM


apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: kabanero-operator-subscription
  namespace: kabanero
spec:
  channel: alpha
  name: kabanero-operator
  source: kabanero-operator-catalog-source
  sourceNamespace: openshift-operator-lifecycle-manager
